src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,38 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor() {
+    super();
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden) {
```

Comment:
[CODE PERFOMANCE] - The `bite` method should avoid unnecessary checks. Since the filtering of the `Animal.alive` array is already handling the removal of dead animals, there's no need to check if `animal.health <= 0` before filtering. You can directly call the filter method after decreasing the health.